File: benchmarks/fixtures/webapp_swift/Sources/webapp/api/AuthHandlers.swift
import Foundation

/// HTTP-facing handlers for authentication endpoints. `handleLogin` is the head
/// of the deep call chain exercised by the trace/call-chain scenarios.
final class AuthHandlers {
    private let service: AuthenticationService
    private let logger: Logger

    init(service: AuthenticationService) {
        self.service = service
        self.logger = getLogger("AuthHandlers")
    }

    /// Handle a POST /login request.
    func handleLogin(email: String, password: String) -> String {
        logger.info("POST /login \(email)")
        do {
            let token = try service.authenticate(email: email, password: password)
            return token.value
        } catch {
            logger.warn("login rejected: \(error)")
            return ""
        }
    }

    /// Handle a POST /logout request.
    func handleLogout(token: String) {
        revokeToken(token)
    }
}

File: benchmarks/fixtures/webapp_kt/src/main/kotlin/webapp/api/AuthHandlers.kt
package webapp.api

import webapp.auth.revokeToken
import webapp.services.AuthenticationService
import webapp.util.Logger
import webapp.util.getLogger

/**
 * HTTP-facing handlers for authentication endpoints. `handleLogin` is the head
 * of the deep call chain exercised by the trace/call-chain scenarios.
 */
class AuthHandlers(private val service: AuthenticationService) {
    private val logger: Logger = getLogger("AuthHandlers")

    /** Handle a POST /login request. */
    fun handleLogin(email: String, password: String): String {
        logger.info("POST /login $email")
        return try {
            val token = service.authenticate(email, password)
            token.value
        } catch (e: Exception) {
            logger.warn("login rejected: $e")
            ""
        }
    }

    /** Handle a POST /logout request. */
    fun handleLogout(token: String) {
        revokeToken(token)
    }
}
